src/nn/prepo.rs

```diff
@@ -14,99 +19,109 @@ extern "C" {
-}
-extern "C" {
+
     #[link_name = "\u{1}_ZN2nn5prepo10PlayReport4SaveERKNS_7account3UidE"]
     pub fn PlayReport_Save1(
         this: *mut PlayReport,
-        arg1: *const root::nn::account::Uid,
+        uid: *const root::nn::account::Uid,
     ) -> root::Result;
-}
-extern "C" {
+
     #[link_name = "\u{1}_ZN2nn5prepo10PlayReportC1Ev"]
     pub fn PlayReport_PlayReport(this: *mut PlayReport);
 }
 impl PlayReport {
     #[inline]
-    pub unsafe fn SetEventId(&mut self, arg1: *const u8) -> root::Result {
-        PlayReport_SetEventId(self, arg1)
+    pub fn SetEventId(&mut self, arg1: *const u8) -> root::Result {
+        unsafe { PlayReport_SetEventId(self, arg1) }
     }
     #[inline]
-    pub unsafe fn SetBuffer(&mut self) -> root::Result {
-        PlayReport_SetBuffer(self)
+    pub fn SetBuffer(&mut self) -> root::Result {
+        unsafe { PlayReport_SetBuffer(self) }
     }
     #[inline]
-    pub unsafe fn Add(
+    pub fn AddLong(
         &mut self,
-        arg1: *const u8,
-        arg2: i64,
+        key: *const u8,
+        value: i64,
     ) -> root::Result {
-        PlayReport_Add(self, arg1, arg2)
+        unsafe { PlayReport_Add(self, key, value) }
     }
     #[inline]
-    pub unsafe fn Add1(
+    pub fn AddDouble(
         &mut self,
-        arg1: *const u8,
-        arg2: f64,
+        key: *const u8,
+        value: f64,
     ) -> root::Result {
-        PlayReport_Add1(self, arg1, arg2)
+        unsafe { PlayReport_Add1(self, key, value) }
     }
     #[inline]
-    pub unsafe fn Add2(
+    pub fn AddString(
         &mut self,
-        arg1: *const u8,
-        arg2: *const u8,
+        key: *const u8,
+        value: *const u8,
+    ) -> root::Result {
+        unsafe { PlayReport_Add2(self, key, value) }
+    }
+    #[inline]
+    pub fn AddAny64BitID(
+        &mut self,
+        key: *const u8,
+        value: Any64BitId,
     ) -> root::Result {
-        PlayReport_Add2(self, arg1, arg2)
+        unsafe { PlayReport_Add3(self, key, value) }
     }
     #[inline]
-    pub unsafe fn Save(&mut self) -> root::Result {
-        PlayReport_Save(self)
+    pub fn Save(&mut self) -> root::Result {
+        unsafe { PlayReport_Save(self) }
     }
     #[inline]
-    pub unsafe fn Save1(
+    pub fn SaveWithAccount(
         &mut self,
-        arg1: *const root::nn::account::Uid,
+        uid: *const root::nn::account::Uid,
     ) -> root::Result {
-        PlayReport_Save1(self, arg1)
+        unsafe { PlayReport_Save1(self, uid) }
     }
     #[inline]
-    pub unsafe fn new() -> Self {
-        let mut __bindgen_tmp = ::core::mem::MaybeUninit::uninit();
-        PlayReport_PlayReport(__bindgen_tmp.as_mut_ptr());
-        __bindgen_tmp.assume_init()
+    pub fn new() -> Self {
+        unsafe {
+            let mut __bindgen_tmp = ::core::mem::MaybeUninit::uninit();
+            PlayReport_PlayReport(__bindgen_tmp.as_mut_ptr());
+            __bindgen_tmp.assume_init()
+        }
```

Comment:
This is completely wrong, you need to make an instance of the PlayReport struct and call the PlayReport constructor on it then assign it a buffer with PlayReport_SetBuffer.

Considering you cannot free the buffer until the PlayReport is destroyed, it'd probably be better to make a new struct that wraps the PlayReport struct alongside its buffer to make it transparent to the caller.

Since calculating the exact size of the buffer would be a pain, feel free to allocate a 0x4000 byte array as it seems to be the maximum size allowed. (I say allocate because if you make it an array, the struct will be massive on the stack and it might cause crashes in some plugins because of the small stack size of ``main()``. So what you might want to do is look into the "Box" type)